docs/setup_devcontainer_environment.mdx

```diff
@@ -0,0 +1,127 @@
+
+First install [Visual Studio code](https://code.visualstudio.com/), than choose one of the next based on your system.
+
+## Setup on Windows
+
+:::tip
+**Quick start guide**
+
+- install [WSL2](https://learn.microsoft.com/en-us/windows/wsl/install) and enable [systemd support](https://devblogs.microsoft.com/commandline/systemd-support-is-now-available-in-wsl/)
+- install [docker engine](https://docs.docker.com/engine/install/) based on your selected WSL2 linux distribution
+- fork and clone the desired code repository
+- [Fork the repository and start the devcontainer](#fork-the-repository-and-start-the-devcontainer)
+
+:::
+
+### Install WSL2
+
+This describes the steps needed based on Windows 10 version 2004 and higher or Windows 11. If you're still running an older version, follow these m[anual installation steps](https://learn.microsoft.com/en-us/windows/wsl/install-manual).
+
+1. run following command in windows terminal which will install needed features to run WSL and installs the default Ubuntu distribution
+
+    ```batch
+    wsl --install
+    ```
+
+2. check WSL version, if version is already 2, skip the "set wsl version" step
+
+    ```batch
+    wsl -l -v
+    ```
+
+3. set WSL version to 2 if neccessary (_replace `<distro name>` by the installed distribution name, see output above_)
+
+    ```batch
+    wsl --set-version <distro name> 2
+    ```
+
+### Enable systemd support in WSL2
+
+Since docker engine is installed as systemd service, we need to ensure, systemd support is enabled in the WSL2
+
+1. start wsl shell by issuing just the `wsl` command in the windows terminal
+2. add the following lines to the `/etc/wsl.conf`, you will need to open the file with root privileges (_eg. `sudo nano /etc/wsl.conf`_)
+
+    ```raw
+    [boot]
+    systemd=true
+    ```
+
+3. shutdown WSL2 with following command on windows terminal
+
+    ```batch
+    wsl --shutdown
+    ```
+
+4. start wsl shell again and run the following command to check that systemd is running
+
+    ```shell
+    systemctl list-units
+    ```
+
+
+### Follow "Setup on linux"
+
+From here you can follow the [Setup on linux](#setup-on-linux) instructions bellow. Use the installed WSL2 distribution (_default is Ubuntu_) when choosing the correct steps to install docker engine.
+
+## Setup on linux
+
+:::tip
+**Quick start guide**
+
+- install [docker engine](https://docs.docker.com/engine/install/) based on your linux distribution
+- fork and clone the desired code repository
+- [Fork the repository and start the devcontainer](#fork-the-repository-and-start-the-devcontainer)
+
+:::
+
+### Install Docker Engine
+
+Since these instructions vary based on your used linux distribution, we only provide a rough overview. For detailed instructions, follow the official [docker engine](https://docs.docker.com/engine/install/) install guide
```

Comment:
```suggestion
Since these instructions vary depending on your Linux distribution, we only provide a rough overview. For detailed instructions, follow the official [Docker engine](https://docs.docker.com/engine/install/) install guide.
```